### PR TITLE
WA-121: Adjust authenticate screen

### PR DESCRIPTION
**Pull request by rmeissner**
Closes WA-121.

Changes proposed in this pull request:
- Adjust design from authenticate screen
- Added select safe screen for ERC67 transactions

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/test/java/pm/gnosis/heimdall/ui/safe/selection/SelectSafeViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,163 @@
+package pm.gnosis.heimdall.ui.safe.selection
+
+import android.content.Context
+import android.content.Intent
+import io.reactivex.Single
+import io.reactivex.functions.Predicate
+import io.reactivex.observers.TestObserver
+import io.reactivex.processors.PublishProcessor
+import io.reactivex.subjects.PublishSubject
```
Comment (fmrsabino) [inline app/src/test/java/pm/gnosis/heimdall/ui/safe/selection/SelectSafeViewModelTest.kt]: Not needed

File: app/src/test/java/pm/gnosis/heimdall/ui/safe/selection/SelectSafeViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        TransactionType.values().forEach {
+            testReviewTransaction(it)
+        }
+    }
+
+    companion object {
+        private val TEST_SAFE = BigInteger.ONE
+        private val TEST_TRANSACTION = Transaction(BigInteger.TEN)
+    }
+}
```
Comment (fmrsabino) [inline app/src/test/java/pm/gnosis/heimdall/ui/safe/selection/SelectSafeViewModelTest.kt]: Newline 😉

File: app/src/main/java/pm/gnosis/heimdall/ui/safe/selection/SelectSafeViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    override fun loadSafes(): Single<List<Safe>> =
+            safeRepository.observeDeployedSafes()
+                    .firstOrError()
+
+    override fun reviewTransaction(safe: BigInteger?, transaction: Transaction): Single<Result<Intent>> =
+            detailRepository.loadTransactionType(transaction)
+                    .map {
+                        safe
+                                ?: throw SimpleLocalizedException(context.getString(R.string.no_safe_selected_error))
```
[the remark this answers is not in the document]
Comment (rmeissner) [reply app/src/main/java/pm/gnosis/heimdall/ui/safe/selection/SelectSafeViewModel.kt]: autoformat :D

File: app/src/test/java/pm/gnosis/heimdall/data/repositories/impls/SimpleTransactionDetailsRepositoryTest.kt [hunk 4]
```diff
@@ -59,15 +62,31 @@ class SimpleTransactionDetailsRepositoryTest {
...
+                // Add safe owners
+                Transaction(BigInteger.ZERO, data = GnosisSafe.AddOwner.encode(Solidity.Address(BigInteger.TEN), Solidity.UInt8(BigInteger.ONE))) to TransactionType.ADD_SAFE_OWNER,
+                // Remove safe owners
+                Transaction(BigInteger.ZERO, data = GnosisSafe.RemoveOwner.encode(Solidity.UInt256(BigInteger.TEN), Solidity.UInt8(BigInteger.ONE))) to TransactionType.REMOVE_SAFE_OWNER,
+                // Replace safe owners
+                Transaction(BigInteger.ZERO, data = GnosisSafe.ReplaceOwner.encode(Solidity.UInt256(BigInteger.TEN), Solidity.Address(BigInteger.TEN))) to TransactionType.REPLACE_SAFE_OWNER,
+                // Unknown data = Generic transaction
+                Transaction(BigInteger.ZERO, data = GnosisSafe.Owners.encode(Solidity.UInt256(BigInteger.ZERO))) to TransactionType.GENERIC
+        )
     }
```
[the remark this answers is not in the document]
Comment (rmeissner) [reply app/src/test/java/pm/gnosis/heimdall/data/repositories/impls/SimpleTransactionDetailsRepositoryTest.kt]: same method under the hood

**Pull request merged**